portfolio-project-matching-app/pages/browseProjects.js

```diff
@@ -107,15 +110,47 @@ const browseProjects = () => {
     }
 
     const onSearch = (searchFilter) => {
-        const updatedProjects = visibleProjects.filter((project) => {
-            return project.name.includes(searchFilter)
-        })
-        setVisibleProjects(updatedProjects)
+        // helper arrays to harbor projects during staging
+        const newHidden = hiddenProjects
+        let newVisible = visibleProjects
+        // loop through hidden projects and potentially add search filter
+        for (const hiddenProject of hiddenProjects) {
+            if (!hiddenProject.project.hasSearchKey(searchFilter)) {
+                hiddenProject.addSearchFilter()
+            }
+        }
+        // loop through visible projects and potentially hide
+        for (const visibleProject of visibleProjects) {
+            if (!visibleProject.hasSearchKey(searchFilter)) {
+                // create HiddenProject object to add to hiddenProjects
+                newHidden.push(FilteredProject.fromSearch(visibleProject, searchFilter))
+                newVisible = newVisible.filter((element) => {
+                    return element.id !== visibleProject.id
+                });
+            }
+        }
+        setVisibleProjects(newVisible)
+        setHiddenProjects(newHidden)
         setIsSearching(true)
     }
 
     const clearSearch = () => {
-        initializeProjects()
+        // helper arrays to harbor projects during staging
+        let newHidden = hiddenProjects
+        const newVisible = visibleProjects
+        // loop through hidden projects and process elements
+        for (const hiddenProject of hiddenProjects) {
+            hiddenProject.removeSearchFilter()
+            // if last remaining filter removed, move from hidden to visible
+            if (hiddenProject.filtersIsEmpty() && !hiddenProject.searchFiltered) {
+                newVisible.push(hiddenProject.project)
+                newHidden = newHidden.filter((element) => {
+                    return element.project.id !== hiddenProject.project.id
+                })
+            }
+        }
+        setVisibleProjects(newVisible)
+        setHiddenProjects(newHidden)
```

Comment:
Similar to `onSearch`, this really just follows the `removeFilter` logic. Helps to keep track of what search/button filters are causing a project to be hidden in case there are multiple filter buttons and a search key being applied.